### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by ishimwepati**
Hello !

Below are the features @coker2019 and @ishimwepati  implemented : 

We used the where method the posts data is being fetched and the comments are loaded for each post causing n+1 problem.
We updated the index.html.erb of posts, where the iteration of five_recent_comments using the includes eager loading method.
We created a new folder views in spec for implementing integration tests for the views using the Capybara gem.
We added test cases that checks the view working from the Capybara.

Your suggestions are most welcome !

image
Screenshot 2023-11-22 at 12 43 58

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-require_relative '../config/environment'
+# require_relative '../config/environment'
+
+require File.expand_path('../config/environment', __dir__)
 # Prevent database truncation if the environment is production
 abort("The Rails environment is running in production mode!") if Rails.env.production?
 require 'rspec/rails'
+require 'capybara/rspec'
+
+Capybara.default_driver = :selenium_chrome
```
Comment (iambenkis) [inline spec/rails_helper.rb:12]: - [x] keep doing the great job

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
+class PostsController < ApplicationController
+  # load_and_authorize_resourcecls
 
-#   def index
-#     @user = User.find(params[:user_id])
-#     @posts = @user.posts.includes(:comments)
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments)
+  end
```
Comment (iambenkis) [inline app/controllers/posts_controller.rb:7]: - [x] Keep doing the great job 🚀

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi Team @ishimwepati , @coker2019 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  Use `incudes` method

## Suggested changes

- [x] Good job adding integration specs for views and fixing n+1 problem, however it is required to [add screenshots of the logs before and after the fix to the PR description](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Do%20not%20use,after%20the%20fix). Kindly add them 👍 

Screenshot 2023-11-21 121358


_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/rails_helper.rb (hunk 1 above)
Comment (ishimwepati) [reply spec/rails_helper.rb:12]: Thanks a lot 👍 !

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (ishimwepati) [reply app/controllers/posts_controller.rb:7]: Thanks a lot 👍 !

**Comment by ishimwepati** [comment]
Screenshot Added as suggested by the reviewer !

**Review by DammyShittu — COMMENTED** [review]
Hi @ishimwepati and @coker2019,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**